client/extensions/zoninator/state/zones/reducer.js

```diff
@@ -17,6 +23,17 @@ export const requesting = createReducer( {}, {
 
 export const items = createReducer( {}, {
 	[ ZONINATOR_UPDATE_ZONES ]: ( state, { siteId, data } ) => ( { ...state, [ siteId ]: data } ),
+	[ ZONINATOR_UPDATE_ZONE ]: ( state, { siteId, data } ) => {
+		const zones = [ ...( state[ siteId ] || [] ) ];
+		const idx = findIndex( zones, { term_id: data.term_id || -1 } );
```

Comment:
I find that when we need something like this in a reducer, it might indicate we should rather have the endpoint return an assoc array keyed by whatever we use to query here, to simplify working with that data in our reducers. (I only had a quick glance, so I might be wrong here.)